src/main/java/com/solvd/elasticsearchmicroservice/kafka/KfConsumerImpl.java

```diff
@@ -1,39 +1,32 @@
 package com.solvd.elasticsearchmicroservice.kafka;
 
 import com.solvd.elasticsearchmicroservice.kafka.event.NoteEvent;
-import com.solvd.elasticsearchmicroservice.kafka.handler.DeleteHandler;
 import com.solvd.elasticsearchmicroservice.kafka.handler.Handler;
-import com.solvd.elasticsearchmicroservice.kafka.handler.PostHandler;
-import com.solvd.elasticsearchmicroservice.kafka.handler.PutHandler;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
 import org.springframework.kafka.annotation.KafkaListener;
 import org.springframework.stereotype.Service;
 
-import java.util.HashMap;
 import java.util.Map;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class KfConsumerImpl implements KfConsumer{
 
-    private final PostHandler postHandler;
-    private final PutHandler putHandler;
-    private final DeleteHandler deleteHandler;
+    private final ApplicationContext context;
+
+    @Bean
+    public Map<String, Handler> handlers(ApplicationContext context) {
```

Comment:
@vpetrovaa redundant logic here

Spring di container will automatically inject all the beans in your map

Reply:
@h1alexbel, removed it in the next pr https://github.com/vpetrovaa/elasticsearch-microservice/pull/4/commits/a16790e32214372ed6016b940a9fa59f66e919ed